std/digest/digest.d

```diff
@@ -1,1171 +1,50 @@
-            result |= r1[i] ^ r2[i];
-    }
-    else static if (hasLength!R1 && hasLength!R2)
-    {
-        // Lengths are the same so we can squeeze out a bit of performance
-        // by not checking if r2 is empty
-        for (; !r1.empty; r1.popFront(), r2.popFront())
-        {
-            result |= r1.front ^ r2.front;
-        }
-    }
-    else
-    {
-        // Generic case, walk both ranges
-        for (; !r1.empty; r1.popFront(), r2.popFront())
-        {
-            if (r2.empty) return false;
-            result |= r1.front ^ r2.front;
-        }
-        if (!r2.empty) return false;
-    }
-
-    return result == 0;
-}
-
-///
-@system pure unittest
-{
-    import std.digest.hmac : hmac;
-    import std.digest.sha : SHA1;
-    import std.string : representation;
-
-    // a typical HMAC data integrity verification
-    auto secret = "A7GZIP6TAQA6OHM7KZ42KB9303CEY0MOV5DD6NTV".representation;
-    auto data = "data".representation;
-
-    string hex1 = data.hmac!SHA1(secret).toHexString;
-    string hex2 = data.hmac!SHA1(secret).toHexString;
-    string hex3 = "data1".representation.hmac!SHA1(secret).toHexString;
-
-    assert( secureEqual(hex1, hex2));
-    assert(!secureEqual(hex1, hex3));
-}
-
-@system pure unittest
-{
-    import std.internal.test.dummyrange : ReferenceInputRange;
-    import std.range : takeExactly;
-    import std.string : representation;
-    import std.utf : byWchar, byDchar;
-
-    {
-        auto hex1 = "02CA3484C375EDD3C0F08D3F50D119E61077".representation;
-        auto hex2 = "02CA3484C375EDD3C0F08D3F50D119E610779018".representation;
-        assert(!secureEqual(hex1, hex2));
-    }
-    {
-        auto hex1 = "02CA3484C375EDD3C0F08D3F50D119E610779018"w.representation;
-        auto hex2 = "02CA3484C375EDD3C0F08D3F50D119E610779018"d.representation;
-        assert(secureEqual(hex1, hex2));
-    }
-    {
-        auto hex1 = "02CA3484C375EDD3C0F08D3F50D119E610779018".byWchar;
-        auto hex2 = "02CA3484C375EDD3C0F08D3F50D119E610779018".byDchar;
-        assert(secureEqual(hex1, hex2));
-    }
-    {
-        auto hex1 = "02CA3484C375EDD3C0F08D3F50D119E61077".byWchar;
-        auto hex2 = "02CA3484C375EDD3C0F08D3F50D119E610779018".byDchar;
-        assert(!secureEqual(hex1, hex2));
-    }
-    {
-        auto hex1 = new ReferenceInputRange!int([0, 1, 2, 3, 4, 5, 6, 7, 8]).takeExactly(9);
-        auto hex2 = new ReferenceInputRange!int([0, 1, 2, 3, 4, 5, 6, 7, 8]).takeExactly(9);
-        assert(secureEqual(hex1, hex2));
-    }
-    {
-        auto hex1 = new ReferenceInputRange!int([0, 1, 2, 3, 4, 5, 6, 7, 8]).takeExactly(9);
-        auto hex2 = new ReferenceInputRange!int([0, 1, 2, 3, 4, 5, 6, 7, 9]).takeExactly(9);
-        assert(!secureEqual(hex1, hex2));
-    }
-}
+static import std.digest;
+
+// Explicitly undocumented. It will be removed in December 2018. @@@DEPRECATED_2018-12@@@
+deprecated("Please use std.digest instead.")
+alias isDigest = std.digest.isDigest;
```

Comment:
> I just tried deprecating a public import, and it has no effect. So, it doesn't look like it's possible to deprecate public imports. Now, what we could do (though it would be a bit messy) would be to change the import to a normal import and then create explicit aliases to all of the symbols that were previously, publicly imported. We could then deprecate the aliases.

Nice idea! I gave this a try and except for `digest` it would work nicely.